xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -182,8 +183,14 @@ public void handleSubchannelState(Subchannel subchannel, ConnectivityStateInfo n
 
     @Override
     public void reset() {
+      List<XdsLocality> toRemove = new ArrayList<>();
       for (XdsLocality locality : localityMap.keySet()) {
+        if (!localityMap.get(locality).isDeactivated()) {
```

Comment:
I understand this is to make sure `loadStatsStore.removeLocality()` will not be called for the deactivated localities, because they have already been `loadStatsStore.removeLocality()`ed in `updateLocalityStore()`, and `loadStatsStore.removeLocality()` doesn't allow calling twice. This has worsened the code readability, and we should probably remove the check in `loadStatsStore.removeLocality()`.

Reply:
After offline discussion with @voidzcy we prefer to only check EDS response changes.